### DISCUSSION TITLE
Impossible to excite TM0 mode with EigenModeSource

### DISCUSSION THREAD

**Discussion by lucasgrjn** [NanoComp/meep#2342] [Q&A]
Hello!

I am currently facing an issue and I cannot figure out where it could come from.
For one of my photonic devices, I would like to make a 1:2 splitter as presented on the tutorial of the [adjoint solver](https://meep.readthedocs.io/en/latest/Python_Tutorials/Adjoint_Solver/), more precisely, [this one](https://nbviewer.org/github/NanoComp/meep/blob/master/python/examples/adjoint_optimization/04-Splitter.ipynb). But I need to guide a TM0 (_Hz-mode_) instead of the TE0 (_Ez-mode_) shown I naively thought I could excite it by changing the variable `eig_band` of the eigenmode source. Unfortunately, this did not work.

So, I did a simple simulation from the example code, just to vary the `eig_band` and to see how it evoluated. What I find stange is the fact it directly excites TM1 but not TM0. (You can find the plotted results below).

Does someone can help me to figure out what happened and why I cannot use the TM0 ?
(Just to be sure, I already did a check using a [mode solver](https://www.siio.eu/oms.html) to verify the guided modes).

Thanks in advance :)
Lucas

![results_eigvals_test](https://user-images.githubusercontent.com/18169686/207263323-68bdfd0d-0bfe-414c-af83-9e49f3984d0c.png)

Here is the used code:
```python
import meep as mp
import numpy as np
from matplotlib import pyplot as plt

seed = 240
np.random.seed(seed)
mp.quiet(quietval=True)
Si = mp.Medium(index=3.4)
SiO2 = mp.Medium(index=1.44)


NUM_TEST_EIGVALS = 6


# Geometry and wavelengths parameters
waveguide_width = 0.5
design_region_width = 2.5
design_region_height = 2.5
arm_separation = 1.0
waveguide_length = 0.5
pml_size = 1.0
resolution = 50

frequencies = 1 / np.linspace(1.5, 1.6, 10)

# Simulation domain
Sx = 2 * pml_size + 2 * waveguide_length + design_region_width
Sy = 2 * pml_size + design_region_height + 0.5
cell_size = mp.Vector3(Sx, Sy)

pml_layers = [mp.PML(pml_size)]

fcen = 1 / 1.56
width = 0.2
fwidth = width * fcen
source_center = [-Sx / 2 + pml_size + waveguide_length / 3, 0, 0]
source_size = mp.Vector3(0, 2, 0)
kpoint = mp.Vector3(1, 0, 0)
src = mp.GaussianSource(frequency=fcen, fwidth=fwidth)

source = [
    mp.EigenModeSource(
        src,
        eig_band=1,
        direction=mp.NO_DIRECTION,
        eig_kpoint=kpoint,
        size=source_size,
        center=source_center,
    )
]

# Object geometry
geometry = [
    mp.Block(
        center=mp.Vector3(),
        material=Si,
        size=mp.Vector3(mp.inf, waveguide_width, 0),
    ),  # left waveguide
]

sim = mp.Simulation(
    cell_size=cell_size,
    boundary_layers=pml_layers,
    geometry=geometry,
    sources=source,
    symmetries=[mp.Mirror(direction=mp.Y)],
    default_material=SiO2,
    resolution=resolution,
)


Ez_fields = []
Hz_fields = []

eig_bands = range(1, NUM_TEST_EIGVALS + 1)
for eig_idx in eig_bands:
    sim.sources[0].eig_band = eig_idx

    sim.fields = None    # prevent the fields from getting deallocated by reset-meep
    sim.reset_meep()

    # Fields register
    out_fields = sim.add_dft_fields(
        # [mp.Ex, mp.Ey, mp.Ez, mp.Hx, mp.Hy, mp.Hz],
        [mp.Ez, mp.Hz],
        np.array([fcen]),
        center=mp.Vector3(*source_center) * -1,
        size=mp.Vector3(0, Sy - 2 * pml_size, 0)
    )

    sim.run(until_after_sources=20)

    Ez_fields.append(sim.get_dft_array(out_fields, mp.Ez, 0))
    Hz_fields.append(sim.get_dft_array(out_fields, mp.Hz, 0))

    print(eig_idx, '/', eig_bands[-1], 'done !')

fig, ax = plt.subplots(2, len(eig_bands),
                       figsize=(len(eig_bands) * 5, 2 * 5))
for idx in range(len(eig_bands)):
    ax[0,idx].plot(np.abs(Ez_fields[idx]))
    ax[0,idx].set_title(f'Measured Ez field for eigval-{idx}')
    ax[1,idx].plot(np.abs(Hz_fields[idx]))
    ax[1,idx].set_title(f'Measured Hz field for eigval-{idx}')

plt.savefig('results_eigvals_test.png')
```

**Comment by smartalecH** [comment]
You have a simulation mirror symmetry. This is only going to support some modes (as described by said symmetry). 

Note that *field* mirror symmetries have a *phase* associated with them too. Change the symmetry sign to -1, and you'll notice you can get the fundamental TM mode finally (but no longer the fundamental TE). 

If you want both, then you need to remove the mirror simulation symmetry. 

(Intuitively you wouldn't even want a mirror geometric symmetry anyway, as it's not possible for a symmetric splitter to do an asymmetric split).
